From fc02d9b7d7f4876ca56c15fb79cc54535d0ad8da Mon Sep 17 00:00:00 2001
From: Steve Barrett <scbarrett@ksu.edu>
Date: Mon, 7 Sep 2015 11:45:37 -0500
Subject: [PATCH] Fixed errors & updated for installing JDK 8_60 on SUSE 13.2.

In keeping with the spirit of SUSE, removed sudo; run as root.

Replaced variable ORACLE_JAVA (/usr/java), which 13.2 seems not to use, with JVM_FOLDER (/usr/lib(64)/jvm). This makes the main link between these two superfluous, so it's gone.

Added alternative update for JDK API documentation, javadocdir.
---
 install-oracle-jdk | 365 +++++++++++++++++++++++++++--------------------------
 1 file changed, 185 insertions(+), 180 deletions(-)

diff --git a/install-oracle-jdk b/install-oracle-jdk
index 2451292..751ff72 100755
--- a/install-oracle-jdk
+++ b/install-oracle-jdk
@@ -13,189 +13,194 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 #
-# Script to automate the installation of the Oracle Java SDK in OpenSuse 13.1
 #
-# Basically this script follows the instructions found on 
-# http://en.opensuse.org/SDB:Installing_Java automating them so that hopefully
-# next time you have to install Oracle JDK you won't have to copy and 
-# paste long commands like these again. 
+# Automates OpenSuse 13.2 installation of Oracle Java 8 SDK.
 #
-# How to use the script:
+# Mostly follows instructions at http://en.opensuse.org/SDB:Installing_Java,
+# automating them so that next time you install an Oracle JDK you won't have
+# to copy and paste such long commands.
 #
-#       1- Download the latest Java rpm SDK from http://www.oracle.com 
-#          (choose 32 or 64 bit according to your platform)
-#       2- Install the SDK by command line (sudo zypper install <rpm file>)
-#          or double clicking on the rpm file
-#       3- Edit this script and adjust the variables below to reflect 
-#           - The latest Java version and revision you are installing
-#           - The location of the java installs - In OpenSuse they are 
-#             located in /usr/lib (32 bit) or /usr/lib64 (64 bits)
-#           - The architecture used i386 (32 bit) or amd64 (64 bit)
-#       4- Run this script from a command line
 #
-# Now, you can select the Oracle Java implementation run these commands, choosing the jdk_Oracle option in each
+# How to use:
+#   1- Download the desired Java rpm JDK from http://www.oracle.com (32 or 64
+#      bit according to your platform), and optionally, the API documentation.
+#   2- Install the JDK by command line: zypper install <rpm file>
+#      or by double clicking on the rpm file.
+#   3- Install API documentation. Optional, but javadocdir update will fail.
+#   4- Adjust the variables below to reflect:
+#      - The Java version and revision being installed.
+#      - The location of the install. In OpenSuse this is either
+#        /usr/lib (32 bit) or /usr/lib64 (64 bits).
+#      - The target architecture i386 (32 bit) or amd64 (64 bit).
+#   5- Run this script as root from a command line
+#   6- Afterwards, select the desired Oracle Java implementation with:
+#     Runtime:         update-alternatives --config java
+#     Compiler:        update-alternatives --config javac
+#     Browser plugin:  update-alternatives --config javaplugin
+#     App launcher:    update-alternatives --config jre_<java version>
+#     Exported JARs:   update-alternatives --config java_sdk_<java version>
+#     [If installed...]
+#     Documentation:   update-alternatives --config javadocdir
 #
-#	Runtime:            sudo update-alternatives --config java
-#	Compiler:           sudo update-alternatives --config javac
-#	Web browser plugin: sudo update-alternatives --config javaplugin
-#	Exported JARs:	    sudo update-alternatives --config java_sdk_<your java version>
 #
-# These are the variables you need to change according to your Java version
-JAVA_VERSION=1.7.0
-JAVA_REVISION=${JAVA_VERSION}_51
-# Change this to /usr/lib on a 32-bit system
-LIB_FOLDER=/usr/lib
-ARCHITECTURE=i386
-
-ORACLE_JAVA=/usr/java/jdk${JAVA_REVISION}
-
-# Compress man pages
-
-for f in ${ORACLE_JAVA}/man/man1/*.1 ; do sudo gzip $(readlink -f $f) ; done ;
-
-# Setup main link
-
-sudo ln -s -T ${ORACLE_JAVA} ${LIB_FOLDER}/jdk_Oracle
-
-# Setup jvm-exports
-
-sudo mkdir ${LIB_FOLDER}/jvm-exports/jdk_Oracle
-cd ${LIB_FOLDER}/jvm-exports/jdk_Oracle
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/rt.jar jaas-${JAVA_VERSION}_Orac.jar
-sudo ln -s jaas-${JAVA_VERSION}_Orac.jar jaas-${JAVA_VERSION}.jar
-sudo ln -s jaas-${JAVA_VERSION}_Orac.jar jaas.jar
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/jce.jar jce-${JAVA_VERSION}_Orac.jar
-sudo ln -s jce-${JAVA_VERSION}_Orac.jar jce-${JAVA_VERSION}.jar
-sudo ln -s jce-${JAVA_VERSION}_Orac.jar jce.jar
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/rt.jar jdbc-stdext-${JAVA_VERSION}_Orac.jar
-sudo ln -s jdbc-stdext-${JAVA_VERSION}_Orac.jar jdbc-stdext-${JAVA_VERSION}.jar
-sudo ln -s jdbc-stdext-${JAVA_VERSION}_Orac.jar jdbc-stdext-3.0.jar
-sudo ln -s jdbc-stdext-${JAVA_VERSION}_Orac.jar jdbc-stdext.jar
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/rt.jar jndi-${JAVA_VERSION}_Orac.jar
-sudo ln -s jndi-${JAVA_VERSION}_Orac.jar jndi-${JAVA_VERSION}.jar
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/rt.jar jndi-cos-${JAVA_VERSION}_Orac.jar
-sudo ln -s jndi-cos-${JAVA_VERSION}_Orac.jar jndi-cos-${JAVA_VERSION}.jar
-sudo ln -s jndi-cos-${JAVA_VERSION}_Orac.jar jndi-cos.jar
-sudo ln -s jndi-${JAVA_VERSION}_Orac.jar jndi.jar
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/rt.jar jndi-ldap-${JAVA_VERSION}_Orac.jar
-sudo ln -s jndi-ldap-${JAVA_VERSION}_Orac.jar jndi-ldap-${JAVA_VERSION}.jar
-sudo ln -s jndi-ldap-${JAVA_VERSION}_Orac.jar jndi-ldap.jar
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/rt.jar jndi-rmi-${JAVA_VERSION}_Orac.jar
-sudo ln -s jndi-rmi-${JAVA_VERSION}_Orac.jar jndi-rmi-${JAVA_VERSION}.jar
-sudo ln -s jndi-rmi-${JAVA_VERSION}_Orac.jar jndi-rmi.jar
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/jsse.jar jsse-${JAVA_VERSION}_Orac.jar
-sudo ln -s jsse-${JAVA_VERSION}_Orac.jar jsse-${JAVA_VERSION}.jar
-sudo ln -s jsse-${JAVA_VERSION}_Orac.jar jsse.jar
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/rt.jar sasl-${JAVA_VERSION}_Orac.jar
-sudo ln -s sasl-${JAVA_VERSION}_Orac.jar sasl-${JAVA_VERSION}.jar
-sudo ln -s sasl-${JAVA_VERSION}_Orac.jar sasl.jar
-
-# Main java
-
-sudo /usr/sbin/update-alternatives --install /usr/bin/java java ${LIB_FOLDER}/jdk_Oracle/bin/java 3 \
-  --slave /usr/share/man/man1/java.1.gz java.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/java.1.gz \
-  --slave ${LIB_FOLDER}/jvm/jre jre ${LIB_FOLDER}/jdk_Oracle/jre \
-  --slave ${LIB_FOLDER}/jvm-exports/jre jre_exports ${LIB_FOLDER}/jvm-exports/jdk_Oracle \
-  --slave /usr/bin/keytool keytool ${LIB_FOLDER}/jdk_Oracle/bin/keytool \
-  --slave /usr/share/man/man1/keytool.1.gz keytool.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/keytool.1.gz \
-  --slave /usr/bin/orbd orbd ${LIB_FOLDER}/jdk_Oracle/bin/orbd \
-  --slave /usr/share/man/man1/orbd.1.gz orbd.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/orbd.1.gz \
-  --slave /usr/bin/policytool policytool ${LIB_FOLDER}/jdk_Oracle/bin/policytool \
-  --slave /usr/share/man/man1/policytool.1.gz policytool.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/policytool.1.gz \
-  --slave /usr/bin/rmid rmid ${LIB_FOLDER}/jdk_Oracle/bin/rmid \
-  --slave /usr/share/man/man1/rmid.1.gz rmid.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/rmid.1.gz \
-  --slave /usr/bin/rmiregistry rmiregistry ${LIB_FOLDER}/jdk_Oracle/bin/rmiregistry \
-  --slave /usr/share/man/man1/rmiregistry.1.gz rmiregistry.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/rmiregistry.1.gz \
-  --slave /usr/bin/servertool servertool ${LIB_FOLDER}/jdk_Oracle/bin/servertool \
-  --slave /usr/share/man/man1/servertool.1.gz servertool.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/servertool.1.gz \
-  --slave /usr/bin/tnameserv tnameserv ${LIB_FOLDER}/jdk_Oracle/bin/tnameserv \
-  --slave /usr/share/man/man1/tnameserv.1.gz tnameserv.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/tnameserv.1.gz
-  
-# Java compiler
-
-sudo /usr/sbin/update-alternatives --install /usr/bin/javac javac ${LIB_FOLDER}/jdk_Oracle/bin/javac 3 \
-  --slave /usr/bin/appletviewer appletviewer ${LIB_FOLDER}/jdk_Oracle/bin/appletviewer \
-  --slave /usr/share/man/man1/appletviewer.1.gz appletviewer.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/appletviewer.1.gz \
-  --slave /usr/bin/apt apt ${LIB_FOLDER}/jdk_Oracle/bin/apt \
-  --slave /usr/share/man/man1/apt.1.gz apt.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/apt.1.gz \
-  --slave /usr/bin/extcheck extcheck ${LIB_FOLDER}/jdk_Oracle/bin/extcheck \
-  --slave /usr/share/man/man1/extcheck.1.gz extcheck.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/extcheck.1.gz \
-  --slave /usr/bin/jar jar ${LIB_FOLDER}/jdk_Oracle/bin/jar \
-  --slave /usr/share/man/man1/jar.1.gz jar.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jar.1.gz \
-  --slave /usr/bin/jarsigner jarsigner ${LIB_FOLDER}/jdk_Oracle/bin/jarsigner \
-  --slave /usr/share/man/man1/jarsigner.1.gz jarsigner.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jarsigner.1.gz \
-  --slave ${LIB_FOLDER}/jvm/java java_sdk ${LIB_FOLDER}/jdk_Oracle \
-  --slave ${LIB_FOLDER}/jvm-exports/java java_sdk_exports ${LIB_FOLDER}/jvm-exports/jdk_Oracle \
-  --slave /usr/share/man/man1/javac.1.gz javac.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/javac.1.gz \
-  --slave /usr/bin/javadoc javadoc ${LIB_FOLDER}/jdk_Oracle/bin/javadoc \
-  --slave /usr/share/man/man1/javadoc.1.gz javadoc.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/javadoc.1.gz \
-  --slave /usr/bin/javah javah ${LIB_FOLDER}/jdk_Oracle/bin/javah \
-  --slave /usr/share/man/man1/javah.1.gz javah.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/javah.1.gz \
-  --slave /usr/bin/javap javap ${LIB_FOLDER}/jdk_Oracle/bin/javap \
-  --slave /usr/share/man/man1/javap.1.gz javap.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/javap.1.gz \
-  --slave /usr/share/man/man1/jcmd.1.gz jcmd.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jcmd.1.gz \
-  --slave /usr/bin/jconsole jconsole ${LIB_FOLDER}/jdk_Oracle/bin/jconsole \
-  --slave /usr/share/man/man1/jconsole.1.gz jconsole.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jconsole.1.gz \
-  --slave /usr/bin/jdb jdb ${LIB_FOLDER}/jdk_Oracle/bin/jdb \
-  --slave /usr/share/man/man1/jdb.1.gz jdb.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jdb.1.gz \
-  --slave /usr/bin/jhat jhat ${LIB_FOLDER}/jdk_Oracle/bin/jhat \
-  --slave /usr/share/man/man1/jhat.1.gz jhat.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jhat.1.gz \
-  --slave /usr/bin/jinfo jinfo ${LIB_FOLDER}/jdk_Oracle/bin/jinfo \
-  --slave /usr/share/man/man1/jinfo.1.gz jinfo.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jinfo.1.gz \
-  --slave /usr/bin/jmap jmap ${LIB_FOLDER}/jdk_Oracle/bin/jmap \
-  --slave /usr/share/man/man1/jmap.1.gz jmap.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jmap.1.gz \
-  --slave /usr/bin/jps jps ${LIB_FOLDER}/jdk_Oracle/bin/jps \
-  --slave /usr/share/man/man1/jps.1.gz jps.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jps.1.gz \
-  --slave /usr/bin/jrunscript jrunscript ${LIB_FOLDER}/jdk_Oracle/bin/jrunscript \
-  --slave /usr/share/man/man1/jrunscript.1.gz jrunscript.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jrunscript.1.gz \
-  --slave /usr/bin/jsadebugd jsadebugd ${LIB_FOLDER}/jdk_Oracle/bin/jsadebugd \
-  --slave /usr/share/man/man1/jsadebugd.1.gz jsadebugd.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jsadebugd.1.gz \
-  --slave /usr/bin/jstack jstack ${LIB_FOLDER}/jdk_Oracle/bin/jstack \
-  --slave /usr/share/man/man1/jstack.1.gz jstack.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jstack.1.gz \
-  --slave /usr/bin/jstat jstat ${LIB_FOLDER}/jdk_Oracle/bin/jstat \
-  --slave /usr/share/man/man1/jstat.1.gz jstat.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jstat.1.gz \
-  --slave /usr/bin/jstatd jstatd ${LIB_FOLDER}/jdk_Oracle/bin/jstatd \
-  --slave /usr/share/man/man1/jstatd.1.gz jstatd.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/jstatd.1.gz \
-  --slave /usr/bin/native2ascii native2ascii ${LIB_FOLDER}/jdk_Oracle/bin/native2ascii \
-  --slave /usr/share/man/man1/native2ascii.1.gz native2ascii.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/native2ascii.1.gz \
-  --slave /usr/bin/pack200 pack200 ${LIB_FOLDER}/jdk_Oracle/bin/pack200 \
-  --slave /usr/share/man/man1/pack200.1.gz pack200.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/pack200.1.gz \
-  --slave /usr/bin/rmic rmic ${LIB_FOLDER}/jdk_Oracle/bin/rmic \
-  --slave /usr/share/man/man1/rmic.1.gz rmic.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/rmic.1.gz \
-  --slave /usr/bin/schemagen schemagen ${LIB_FOLDER}/jdk_Oracle/bin/schemagen \
-  --slave /usr/share/man/man1/schemagen.1.gz schemagen.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/schemagen.1.gz \
-  --slave /usr/bin/serialver serialver ${LIB_FOLDER}/jdk_Oracle/bin/serialver \
-  --slave /usr/share/man/man1/serialver.1.gz serialver.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/serialver.1.gz \
-  --slave /usr/bin/unpack200 unpack200 ${LIB_FOLDER}/jdk_Oracle/bin/unpack200 \
-  --slave /usr/share/man/man1/unpack200.1.gz unpack200.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/unpack200.1.gz \
-  --slave /usr/bin/wsgen wsgen ${LIB_FOLDER}/jdk_Oracle/bin/wsgen \
-  --slave /usr/share/man/man1/wsgen.1.gz wsgen.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/wsgen.1.gz \
-  --slave /usr/bin/wsimport wsimport ${LIB_FOLDER}/jdk_Oracle/bin/wsimport \
-  --slave /usr/share/man/man1/wsimport.1.gz wsimport.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/wsimport.1.gz \
-  --slave /usr/bin/xjc xjc ${LIB_FOLDER}/jdk_Oracle/bin/xjc \
-  --slave /usr/share/man/man1/xjc.1.gz xjc.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/xjc.1.gz  
-
-# Web browser plugin
-
-sudo /usr/sbin/update-alternatives --install ${LIB_FOLDER}/browser-plugins/javaplugin.so javaplugin ${LIB_FOLDER}/jdk_Oracle/jre/lib/${ARCHITECTURE}/libnpjp2.so 3 \
-  --slave /usr/bin/javaws javaws ${LIB_FOLDER}/jdk_Oracle/jre/bin/javaws \
-  --slave /usr/share/man/man1/javaws.1 javaws.1 ${LIB_FOLDER}/jdk_Oracle/man/man1/javaws.1.gz
-  
-# Java application launcher
-
-sudo /usr/sbin/update-alternatives --install ${LIB_FOLDER}/jvm/jre-${JAVA_VERSION} jre_${JAVA_VERSION} ${LIB_FOLDER}/jdk_Oracle/jre 3  \
-  --slave ${LIB_FOLDER}/jvm-exports/jre-${JAVA_VERSION} jre_${JAVA_VERSION}_exports ${LIB_FOLDER}/jvm-exports/jdk_Oracle
-
-# Exported JARs
-  
-sudo /usr/sbin/update-alternatives --install ${LIB_FOLDER}/jvm/java-${JAVA_VERSION} java_sdk_${JAVA_VERSION} ${LIB_FOLDER}/jdk_Oracle 3 \
-  --slave ${LIB_FOLDER}/jvm-exports/java-${JAVA_VERSION} java_sdk_${JAVA_VERSION}_exports ${LIB_FOLDER}/jvm-exports/jdk_Oracle
-  
-# Control panel
-
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/bin/jcontrol /usr/bin/
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/desktop/icons/hicolor/16x16/apps/sun-jcontrol.png /usr/share/icons/hicolor/16x16/apps/
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/desktop/icons/hicolor/48x48/apps/sun-jcontrol.png /usr/share/icons/hicolor/48x48/apps/
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/desktop/icons/LowContrast/16x16/apps/sun-jcontrol.png /usr/share/icons/locolor/16x16/apps/
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/desktop/icons/LowContrast/48x48/apps/sun-jcontrol.png /usr/share/icons/locolor/48x48/apps/
-sudo ln -s ${LIB_FOLDER}/jdk_Oracle/jre/lib/desktop/applications/sun_java.desktop /usr/share/applications
+# Set these variables according to the Java version to be installed:
+JAVA_VERSION=1.8.0               # JAVA_VERSION  = 1.8.0
+JAVA_REVISION=${JAVA_VERSION}_60 # JAVA_REVISION = 1.8.0_60
+ARCHITECTURE=amd64               # ARCHITECTURE  = i386 or amd64
+PRIORITY=18060                   # PRIORITY      = <your priority>
+LIB_FOLDER=/usr/lib64            # LIB_FOLDER    = /usr/lib(64)
+                                 # JVM_FOLDER    = /usr/lib(64)/jvm/jdk1.8.0_60
+JVM_FOLDER=${LIB_FOLDER}/jvm/jdk${JAVA_REVISION}
+
+
+# Compress the java man pages.
+for f in ${JVM_FOLDER}/man/man1/*.1 ; do gzip $(readlink -f $f) ; done ;
+
+
+# Setup the jvm-exports.
+mkdir -p ${LIB_FOLDER}/jvm-exports/jdk${JAVA_REVISION}
+cd ${LIB_FOLDER}/jvm-exports/jdk${JAVA_REVISION}
+ln -s ${JVM_FOLDER}/jre/lib/rt.jar jaas-${JAVA_REVISION}.jar
+ln -s jaas-${JAVA_REVISION}.jar jaas-${JAVA_VERSION}.jar
+ln -s jaas-${JAVA_REVISION}.jar jaas.jar
+ln -s ${JVM_FOLDER}/jre/lib/jce.jar jce-${JAVA_REVISION}.jar
+ln -s jce-${JAVA_REVISION}.jar jce-${JAVA_VERSION}.jar
+ln -s jce-${JAVA_REVISION}.jar jce.jar
+ln -s ${JVM_FOLDER}/jre/lib/rt.jar jdbc-stdext-${JAVA_REVISION}.jar
+ln -s jdbc-stdext-${JAVA_REVISION}.jar jdbc-stdext-${JAVA_VERSION}.jar
+ln -s jdbc-stdext-${JAVA_REVISION}.jar jdbc-stdext-3.0.jar
+ln -s jdbc-stdext-${JAVA_REVISION}.jar jdbc-stdext.jar
+ln -s ${JVM_FOLDER}/jre/lib/rt.jar jndi-${JAVA_REVISION}.jar
+ln -s jndi-${JAVA_REVISION}.jar jndi-${JAVA_VERSION}.jar
+ln -s jndi-${JAVA_REVISION}.jar jndi.jar
+ln -s ${JVM_FOLDER}/jre/lib/rt.jar jndi-cos-${JAVA_REVISION}.jar
+ln -s jndi-cos-${JAVA_REVISION}.jar jndi-cos-${JAVA_VERSION}.jar
+ln -s jndi-cos-${JAVA_REVISION}.jar jndi-cos.jar
+ln -s ${JVM_FOLDER}/jre/lib/rt.jar jndi-ldap-${JAVA_REVISION}.jar
+ln -s jndi-ldap-${JAVA_REVISION}.jar jndi-ldap-${JAVA_VERSION}.jar
+ln -s jndi-ldap-${JAVA_REVISION}.jar jndi-ldap.jar
+ln -s ${JVM_FOLDER}/jre/lib/rt.jar jndi-rmi-${JAVA_REVISION}.jar
+ln -s jndi-rmi-${JAVA_REVISION}.jar jndi-rmi-${JAVA_VERSION}.jar
+ln -s jndi-rmi-${JAVA_REVISION}.jar jndi-rmi.jar
+ln -s ${JVM_FOLDER}/jre/lib/jsse.jar jsse-${JAVA_REVISION}.jar
+ln -s jsse-${JAVA_REVISION}.jar jsse-${JAVA_VERSION}.jar
+ln -s jsse-${JAVA_REVISION}.jar jsse.jar
+ln -s ${JVM_FOLDER}/jre/lib/rt.jar sasl-${JAVA_REVISION}.jar
+ln -s sasl-${JAVA_REVISION}.jar sasl-${JAVA_VERSION}.jar
+ln -s sasl-${JAVA_REVISION}.jar sasl.jar
+
+
+# >>> java <<< Symbolic links to java JRE commands.
+/usr/sbin/update-alternatives --install /usr/bin/java java ${JVM_FOLDER}/bin/java ${PRIORITY} \
+  --slave ${LIB_FOLDER}/jvm/jre jre ${JVM_FOLDER}/jre \
+  --slave ${LIB_FOLDER}/jvm-exports/jre jre_exports ${LIB_FOLDER}/jvm-exports/jdk${JAVA_REVISION} \
+  --slave /usr/share/man/man1/java.1.gz java.1.gz ${JVM_FOLDER}/man/man1/java.1.gz \
+  --slave /usr/bin/keytool keytool ${JVM_FOLDER}/bin/keytool \
+  --slave /usr/share/man/man1/keytool.1.gz keytool.1.gz ${JVM_FOLDER}/man/man1/keytool.1.gz \
+  --slave /usr/bin/orbd orbd ${JVM_FOLDER}/bin/orbd \
+  --slave /usr/share/man/man1/orbd.1.gz orbd.1.gz ${JVM_FOLDER}/man/man1/orbd.1.gz \
+  --slave /usr/bin/policytool policytool ${JVM_FOLDER}/bin/policytool \
+  --slave /usr/share/man/man1/policytool.1.gz policytool.1.gz ${JVM_FOLDER}/man/man1/policytool.1.gz \
+  --slave /usr/bin/rmid rmid ${JVM_FOLDER}/bin/rmid \
+  --slave /usr/share/man/man1/rmid.1.gz rmid.1.gz ${JVM_FOLDER}/man/man1/rmid.1.gz \
+  --slave /usr/bin/rmiregistry rmiregistry ${JVM_FOLDER}/bin/rmiregistry \
+  --slave /usr/share/man/man1/rmiregistry.1.gz rmiregistry.1.gz ${JVM_FOLDER}/man/man1/rmiregistry.1.gz \
+  --slave /usr/bin/servertool servertool ${JVM_FOLDER}/bin/servertool \
+  --slave /usr/share/man/man1/servertool.1.gz servertool.1.gz ${JVM_FOLDER}/man/man1/servertool.1.gz \
+  --slave /usr/bin/tnameserv tnameserv ${JVM_FOLDER}/bin/tnameserv \
+  --slave /usr/share/man/man1/tnameserv.1.gz tnameserv.1.gz ${JVM_FOLDER}/man/man1/tnameserv.1.gz
+
+
+# >>> javac <<< Symbolic links to java JDK commands.
+/usr/sbin/update-alternatives --install /usr/bin/javac javac ${JVM_FOLDER}/bin/javac ${PRIORITY} \
+  --slave ${LIB_FOLDER}/jvm/java java_sdk ${JVM_FOLDER} \
+  --slave ${LIB_FOLDER}/jvm-exports/java java_sdk_exports ${LIB_FOLDER}/jvm-exports/jdk${JAVA_REVISION} \
+  --slave /usr/share/man/man1/javac.1.gz javac.1.gz ${JVM_FOLDER}/man/man1/javac.1.gz \
+  --slave /usr/bin/appletviewer appletviewer ${JVM_FOLDER}/bin/appletviewer \
+  --slave /usr/share/man/man1/appletviewer.1.gz appletviewer.1.gz ${JVM_FOLDER}/man/man1/appletviewer.1.gz \
+  # Part of Java 7, removed from 8.
+  #--slave /usr/bin/apt apt ${LIB_FOLDER}/jdk_Oracle/bin/apt \
+  #--slave /usr/share/man/man1/apt.1.gz apt.1.gz ${LIB_FOLDER}/jdk_Oracle/man/man1/apt.1.gz \
+  --slave /usr/bin/extcheck extcheck ${JVM_FOLDER}/bin/extcheck \
+  --slave /usr/share/man/man1/extcheck.1.gz extcheck.1.gz ${JVM_FOLDER}/man/man1/extcheck.1.gz \
+  --slave /usr/bin/jar jar ${JVM_FOLDER}/bin/jar \
+  --slave /usr/share/man/man1/jar.1.gz jar.1.gz ${JVM_FOLDER}/man/man1/jar.1.gz \
+  --slave /usr/bin/jarsigner jarsigner ${JVM_FOLDER}/bin/jarsigner \
+  --slave /usr/share/man/man1/jarsigner.1.gz jarsigner.1.gz ${JVM_FOLDER}/man/man1/jarsigner.1.gz \
+  --slave /usr/bin/javadoc javadoc ${JVM_FOLDER}/bin/javadoc \
+  --slave /usr/share/man/man1/javadoc.1.gz javadoc.1.gz ${JVM_FOLDER}/man/man1/javadoc.1.gz \
+  --slave /usr/bin/javah javah ${JVM_FOLDER}/bin/javah \
+  --slave /usr/share/man/man1/javah.1.gz javah.1.gz ${JVM_FOLDER}/man/man1/javah.1.gz \
+  --slave /usr/bin/javap javap ${JVM_FOLDER}/bin/javap \
+  --slave /usr/share/man/man1/javap.1.gz javap.1.gz ${JVM_FOLDER}/man/man1/javap.1.gz \
+  --slave /usr/bin/jcmd jcmd ${JVM_FOLDER}/bin/jcmd \
+  --slave /usr/share/man/man1/jcmd.1.gz jcmd.1.gz ${JVM_FOLDER}/man/man1/jcmd.1.gz \
+  --slave /usr/bin/jconsole jconsole ${JVM_FOLDER}/bin/jconsole \
+  --slave /usr/share/man/man1/jconsole.1.gz jconsole.1.gz ${JVM_FOLDER}/man/man1/jconsole.1.gz \
+  --slave /usr/bin/jdb jdb ${JVM_FOLDER}/bin/jdb \
+  --slave /usr/share/man/man1/jdb.1.gz jdb.1.gz ${JVM_FOLDER}/man/man1/jdb.1.gz \
+  --slave /usr/bin/jhat jhat ${JVM_FOLDER}/bin/jhat \
+  --slave /usr/share/man/man1/jhat.1.gz jhat.1.gz ${JVM_FOLDER}/man/man1/jhat.1.gz \
+  --slave /usr/bin/jinfo jinfo ${JVM_FOLDER}/bin/jinfo \
+  --slave /usr/share/man/man1/jinfo.1.gz jinfo.1.gz ${JVM_FOLDER}/man/man1/jinfo.1.gz \
+  --slave /usr/bin/jmap jmap ${JVM_FOLDER}/bin/jmap \
+  --slave /usr/share/man/man1/jmap.1.gz jmap.1.gz ${JVM_FOLDER}/man/man1/jmap.1.gz \
+  --slave /usr/bin/jps jps ${JVM_FOLDER}/bin/jps \
+  --slave /usr/share/man/man1/jps.1.gz jps.1.gz ${JVM_FOLDER}/man/man1/jps.1.gz \
+  --slave /usr/bin/jrunscript jrunscript ${JVM_FOLDER}/bin/jrunscript \
+  --slave /usr/share/man/man1/jrunscript.1.gz jrunscript.1.gz ${JVM_FOLDER}/man/man1/jrunscript.1.gz \
+  --slave /usr/bin/jsadebugd jsadebugd ${JVM_FOLDER}/bin/jsadebugd \
+  --slave /usr/share/man/man1/jsadebugd.1.gz jsadebugd.1.gz ${JVM_FOLDER}/man/man1/jsadebugd.1.gz \
+  --slave /usr/bin/jstack jstack ${JVM_FOLDER}/bin/jstack \
+  --slave /usr/share/man/man1/jstack.1.gz jstack.1.gz ${JVM_FOLDER}/man/man1/jstack.1.gz \
+  --slave /usr/bin/jstat jstat ${JVM_FOLDER}/bin/jstat \
+  --slave /usr/share/man/man1/jstat.1.gz jstat.1.gz ${JVM_FOLDER}/man/man1/jstat.1.gz \
+  --slave /usr/bin/jstatd jstatd ${JVM_FOLDER}/bin/jstatd \
+  --slave /usr/share/man/man1/jstatd.1.gz jstatd.1.gz ${JVM_FOLDER}/man/man1/jstatd.1.gz \
+  --slave /usr/bin/native2ascii native2ascii ${JVM_FOLDER}/bin/native2ascii \
+  --slave /usr/share/man/man1/native2ascii.1.gz native2ascii.1.gz ${JVM_FOLDER}/man/man1/native2ascii.1.gz \
+  --slave /usr/bin/pack200 pack200 ${JVM_FOLDER}/bin/pack200 \
+  --slave /usr/share/man/man1/pack200.1.gz pack200.1.gz ${JVM_FOLDER}/man/man1/pack200.1.gz \
+  --slave /usr/bin/rmic rmic ${JVM_FOLDER}/bin/rmic \
+  --slave /usr/share/man/man1/rmic.1.gz rmic.1.gz ${JVM_FOLDER}/man/man1/rmic.1.gz \
+  --slave /usr/bin/schemagen schemagen ${JVM_FOLDER}/bin/schemagen \
+  --slave /usr/share/man/man1/schemagen.1.gz schemagen.1.gz ${JVM_FOLDER}/man/man1/schemagen.1.gz \
+  --slave /usr/bin/serialver serialver ${JVM_FOLDER}/bin/serialver \
+  --slave /usr/share/man/man1/serialver.1.gz serialver.1.gz ${JVM_FOLDER}/man/man1/serialver.1.gz \
+  --slave /usr/bin/unpack200 unpack200 ${JVM_FOLDER}/bin/unpack200 \
+  --slave /usr/share/man/man1/unpack200.1.gz unpack200.1.gz ${JVM_FOLDER}/man/man1/unpack200.1.gz \
+  --slave /usr/bin/wsgen wsgen ${JVM_FOLDER}/bin/wsgen \
+  --slave /usr/share/man/man1/wsgen.1.gz wsgen.1.gz ${JVM_FOLDER}/man/man1/wsgen.1.gz \
+  --slave /usr/bin/wsimport wsimport ${JVM_FOLDER}/bin/wsimport \
+  --slave /usr/share/man/man1/wsimport.1.gz wsimport.1.gz ${JVM_FOLDER}/man/man1/wsimport.1.gz \
+  --slave /usr/bin/xjc xjc ${JVM_FOLDER}/bin/xjc \
+  --slave /usr/share/man/man1/xjc.1.gz xjc.1.gz ${JVM_FOLDER}/man/man1/xjc.1.gz
+
+
+# >>> javaplugin <<< Symbolic links to Web browser plugin commands.
+/usr/sbin/update-alternatives --install ${LIB_FOLDER}/browser-plugins/javaplugin.so javaplugin ${JVM_FOLDER}/jre/lib/${ARCHITECTURE}/libnpjp2.so ${PRIORITY} \
+  --slave /usr/bin/javaws javaws ${JVM_FOLDER}/jre/bin/javaws \
+  --slave /usr/share/man/man1/javaws.1 javaws.1 ${JVM_FOLDER}/man/man1/javaws.1.gz
+
+
+# >>> java_jre_1.8.0 <<< Symbolic links to java 1.8.0 JRE commands.
+/usr/sbin/update-alternatives --install ${LIB_FOLDER}/jvm/jre-${JAVA_VERSION} jre_${JAVA_VERSION} ${JVM_FOLDER}/jre ${PRIORITY} \
+  --slave ${LIB_FOLDER}/jvm-exports/jre-${JAVA_VERSION} jre_${JAVA_VERSION}_exports ${LIB_FOLDER}/jvm-exports/jdk${JAVA_REVISION}
+
+
+# >>> java_sdk_1.8.0 <<< Symbolic links to java 1.8.0 JDK commands.
+/usr/sbin/update-alternatives --install ${LIB_FOLDER}/jvm/java-${JAVA_VERSION} java_sdk_${JAVA_VERSION} ${JVM_FOLDER} ${PRIORITY} \
+  --slave ${LIB_FOLDER}/jvm-exports/java-${JAVA_VERSION} java_sdk_${JAVA_VERSION}_exports ${LIB_FOLDER}/jvm-exports/jdk${JAVA_REVISION}
+
+
+# >>> javadocdir <<< Symbolic links to java API documentation.
+/usr/sbin/update-alternatives --install /usr/share/javadoc/java javadocdir /usr/share/javadoc/jdk${JAVA_REVISION}/api ${PRIORITY}
+
+
+# Expose Sun's Java control panel and various icons.
+ln -s ${JVM_FOLDER}/jre/bin/jcontrol /usr/bin/
+ln -s ${JVM_FOLDER}/jre/lib/desktop/icons/hicolor/16x16/apps/sun-jcontrol.png /usr/share/icons/hicolor/16x16/apps/
+ln -s ${JVM_FOLDER}/jre/lib/desktop/icons/hicolor/48x48/apps/sun-jcontrol.png /usr/share/icons/hicolor/48x48/apps/
+ln -s ${JVM_FOLDER}/jre/lib/desktop/icons/LowContrast/16x16/apps/sun-jcontrol.png /usr/share/icons/locolor/16x16/apps/
+ln -s ${JVM_FOLDER}/jre/lib/desktop/icons/LowContrast/48x48/apps/sun-jcontrol.png /usr/share/icons/locolor/48x48/apps/
+ln -s ${JVM_FOLDER}/jre/lib/desktop/applications/sun_java.desktop /usr/share/applications
-- 
2.1.4

